addons/sale_purchase/models/sale_order_line.py

```diff
@@ -102,14 +102,16 @@ def _purchase_increase_ordered_qty(self, new_qty, origin_values):
                 in UoM of SO line.
             :param origin_values: map from sale line id to old value for the ordered quantity (dict)
         """
+        increased_qty_line = self.env['sale.order.line']
         for line in self:
             last_purchase_line = self.env['purchase.order.line'].search([('sale_line_id', '=', line.id)], order='create_date DESC', limit=1)
             if last_purchase_line.state in ['draft', 'sent', 'to approve']:  # update qty for draft PO lines
                 quantity = line.product_uom._compute_quantity(new_qty, last_purchase_line.product_uom)
                 last_purchase_line.write({'product_qty': quantity})
             elif last_purchase_line.state in ['purchase', 'done', 'cancel']:  # create new PO, by forcing the quantity as the difference from SO line
                 quantity = line.product_uom._compute_quantity(new_qty - origin_values.get(line.id, 0.0), last_purchase_line.product_uom)
-                line._purchase_service_create(quantity=quantity)
+                increased_qty_line += line
+        increased_qty_line._purchase_service_create(quantity=quantity)
```

Comment:
Could you explain your logic here ? i don't get it :thinking: 
You're creating a recordset of sale line, on which you use the method _purchase_service_create instead of calling the method on each line individually. Ok. But doing so, you're using the same quantity for every line no ? I suppose there are cases where the quantity are different though.... 
From what i see, it does not help reaching the task's goal, in addition to creating incorrect data. I suppose i'm missing something, but i don't get what :face_with_spiral_eyes:

Reply:
i did this to reduce the number of function calls by using the _purchase_service_create method on a recordset instead of calling it separately for each line, currently we are getting only one record in self at once but indeed it will nt gives correct quantity when we get multiple record in self at a time.